cli/src/complete.rs

```diff
@@ -43,15 +45,113 @@ pub fn local_bookmarks() -> Vec<CompletionCandidate> {
     })
 }
 
+pub fn tracked_bookmarks() -> Vec<CompletionCandidate> {
+    with_jj(|jj, _| {
+        let output = jj
+            .build()
+            .arg("bookmark")
+            .arg("list")
+            .arg("--tracked")
+            .arg("--template")
+            .arg(r#"if(!remote, name ++ "\n")"#)
+            .output()
+            .map_err(user_error)?;
+
+        Ok(String::from_utf8_lossy(&output.stdout)
+            .lines()
+            .filter(|bookmark| !bookmark.ends_with("@git"))
+            .map(CompletionCandidate::new)
+            .collect())
+    })
+}
+
+pub fn untracked_bookmarks() -> Vec<CompletionCandidate> {
+    with_jj(|jj, config| {
+        let tracked = jj
+            .build()
+            .arg("bookmark")
+            .arg("list")
+            .arg("--tracked")
+            .arg("--template")
+            .arg(r#"if(remote, name ++ "@" ++ remote ++ "\n")"#)
+            .output()
+            .map_err(user_error)?;
+        let tracked = String::from_utf8_lossy(&tracked.stdout);
+        let tracked: Vec<_> = tracked.lines().collect();
+
+        let output = jj
+            .build()
+            .arg("bookmark")
+            .arg("list")
+            .arg("--all-remotes")
+            .arg("--template")
+            .arg(r#"if(remote, name ++ "@" ++ remote ++ "\n")"#)
+            .output()
+            .map_err(user_error)?;
```

Comment:
maybe we can abuse template `if(remote && !tracked, ..)` to filter out tracked bookmarks